src/LineageHelper.ts

```diff
@@ -36,28 +47,27 @@ interface EdgeId {
 }
 
 /**
- * The element - table, column, edge - that is in focus
+ * A specific element in the lineage graph
  */
 export type ElementId = TableId | ColumnId | EdgeId;
 
-export type MatchedElement<TableData, ColumnData> = LineageElement<TableData, ColumnData> | ColumnReference<ColumnData>;
-
+/**
+ * Reference to a column
+ * Includes the id of the table it is part of
+ */
 export interface ColumnReference<ColumnData> {
   type: "column";
   tableId: string;
   column: Column<ColumnData>;
 }
 
-export const toElementId = <TableData, ColumnData>(element: MatchedElement<TableData, ColumnData>): ElementId => {
-  switch (element.type) {
-    case "table":
-      return { type: "table", tableId: element.id };
-    case "column":
-      return { type: "column", tableId: element.tableId, columnId: element.column.id };
-    case "edge":
-      return element;
-  }
-};
+type ConnectedElement<TableData, ColumnData> = Table<TableData, ColumnData> | ColumnReference<ColumnData> | Edge;
+
+export interface ConnectedElements<TableData, ColumnData> {
+  tables: Table<TableData, ColumnData>[];
+  columns: ColumnReference<ColumnData>[];
+  edges: Edge[];
+}
```

Comment:
New return type for connected elements. We need to differentiate tables and columns here.